index.php

```diff
@@ -250,138 +250,215 @@
 }
 
-function process_data_field(&$data, &$where, &$params, $db_field, $field_name, $comparison, $start_group = false, $force_and = false, $end_group = false) {
-    if(isset($data->{"$field_name"}) && $data->{"$field_name"}) {
-        add_condition_connectors($data, $where, $params, $force_and);
-        $params[$field_name] = $data->{"$field_name"}.($comparison == "LIKE" ? "%" : "");
+/**
+ * Builds a SQL query for case insensitive searching from form data.
+ * @param object $data Form data stored in session.
+ * @param string $where Where clause.
+ * @param array $params Search parameter. Typically stores user ID and cohort
+ * ID in other uses.
+ * @param string $dbfield Value of a database field.
+ * @param string $fieldname Name of a database field.
+ * @param string $comparison Comparison operator or a LIKE operator.
+ * @param string|bool $startgroup Could be used to append ")" to a WHERE
+ * clause.
+ * @param string $forceand A nonempty value would convert an OR operator
+ * to AND.
+ * @param string|bool $endgroup Could be used to append ")" to a WHERE clause.
+ */
+function process_data_field(&$data, &$where, &$params, $dbfield, $fieldname,
+    $comparison, $startgroup, $forceand, $endgroup = null) {
+
+    $startgroup = set_false_if_empty($startgroup);
+    $forceand = set_false_if_empty($forceand);
+    $endgroup = set_false_if_empty($endgroup);
 
-        if($start_group) {
+    if (isset($data->{"$fieldname"}) && $data->{"$fieldname"}) {
+        add_condition_connectors($data, $where, $params, $forceand);
+        $params[$fieldname] = $data->{"$fieldname"} . ($comparison == "LIKE" ? "%" : "");
+
+        if ($startgroup) {
             $where .= "(";
         }
 
         $colon = ":";
 
-        // If the search parameter is a string, make it case insensitive
-        if(preg_match("/[a-z]/i", $params[$field_name])) {
-            $db_field = "LOWER($db_field)";
-            $field_name = "LOWER($colon$field_name)";
+        // If the search parameter is a string, make it case insensitive.
+        if (preg_match("/[a-z]/i", $params[$fieldname])) {
+            $dbfield = "LOWER($dbfield)";
+            $fieldname = "LOWER($colon$fieldname)";
             $colon = "";
         }
 
-        $where .= "$db_field $comparison $colon$field_name";
+        $where .= "$dbfield $comparison $colon$fieldname";
 
-        if($end_group) {
+        if ($endgroup) {
             $where .= ")";
         }
     }
 }
 
+/**
+ * Assigns a boolean false value to any empty parameter.
+ *
+ * Passing parameters here from the top of another function allows that
+ * function to set a value to FALSE by default without failing the test for
+ * single responsibility violations. This is important, because setting
+ * default values on the first line of a function definition does fail
+ * the test for single responsibility violations.
+ *
+ * @param string|int|float|bool|array|NULL $param
+ * @return bool
+ */
+function set_false_if_empty($param) {
```

Comment:
please remove this function and just use PHP's Ternary operator (example mentioned above.)